test/models/rule_test.rb

```diff
@@ -59,4 +59,25 @@ class RuleTest < ActiveSupport::TestCase
     )
     assert_equal [rules(:one)], Rule.canonical
   end
+
+  test 'rule generates remediation issue id' do
+    rule = rules(:one)
+    # Append non-canonical profile to the rule
+    profiles(:one).update!(
+      ref_id: 'xccdf_org.ssgproject.content_profile_profile1',
+      parent_profile: profiles(:two)
+    )
+    rule.profiles << profiles(:one)
+
+    assert_equal rule.remediation_issue_id, 'ssg:rhel7|profile1|MyStringOne'
+  end
+
+  test 'rule remediation issue id raises for two non-unique profiles' do
+    rule = rules(:one)
+    rule.profiles = [profiles(:one), profiles(:two)]
+
+    assert_raises StandardError do
+      rule.remediation_issue_id
+    end
+  end
```

Comment:
I think we should just grab the first profile like we did before. It's unclear if the remediations are the same between profiles for the same rule, but I believe that's the case. Essentially, this would break something that's working today, even if it's not ideal. RHICOMPL-648 should resolve these sort of issues with our remediations integration.